changelog/front-commerce-3.5-2.30.mdx

```diff
@@ -0,0 +1,198 @@
+will have to remove the route from your project as documented in the migration
+guide for your specific version.
+
+:::
+
+## Server-side tracking
+
+With the decline of 3rd-party cookies, it is essential to have a server-side
+tracking solution to keep track of your customers' behavior and understand their
+needs.
+
+Today's Front-Commerce releases introduce different options for your projects:
+
+- **2.x (and 3.x) projects** can leverage tools such as
+  [Server-side Tag Manager](https://developers.google.com/tag-platform/tag-manager/server-side)
+  or commercial solutions such as [Addingwell](https://www.addingwell.com/) to
+  send tracking events to 3rd-party services from a server-side context hosted
+  on a subdomain. We updated our documentation to explain how to set up these
+  tools with Front-Commerce's existing analytics stack.
+- **3.5+ projects** can additionally benefit from
+  [Server-Side events](/docs/3.x/category/server-side-events) to use our new
+  server-side tracking solution that is fully integrated with Front-Commerce and
+  server-side [Analytics plugins](https://getanalytics.io/plugins/). We've
+  [added support for Google Analytics Measurement Protocol](/docs/3.x/guides/server-side-events/integrations/google-analytics)
+  commerce events, and are looking for feedbacks to improve and extend this
+  feature.
+
+:::note Learn more
+
+For more context on server-side tracking, you can read our recent editorial
+content on the subject:
+
+- [Understanding Server-Side Tracking](https://www.front-commerce.com/understanding-server-side-tracking/)
+- [Exploring Server-Side Tracking and Server-Sent Events](https://www.front-commerce.com/exploring-server-side-tracking-and-server-sent-events/)
+- [Client-side Tracking vs. Server-side Tracking: Navigating The Challenges and Solutions](https://www.front-commerce.com/client-side-tracking-vs-server-side-tracking-navigating-the-challenges-and-solutions/)
+- [Exploring Server-Side Tracking with Front-Commerce](https://www.front-commerce.com/exploring-server-side-tracking-with-front-commerce/)
+- [Understanding Server-Side Tracking for Enhanced Data Collection](https://www.front-commerce.com/understanding-server-side-tracking-for-enhanced-data-collection/)
+
+:::
+
+## Dynamic Routing in Front-Commerce 3.5
+
+Front-Commerce 3.5 introduces dynamic routing, a feature that brings more
+flexibility than ever to your Remix application. It allows you to customize your
+URL structure dynamically based on your data, without sacrificing performance or
+features.
+
+**This feature is unique in the Remix ecosystem and closes the gap with the 2.x
+versions of Front-Commerce.** We took the time to build it right, and we are
+looking forward to seeing how we can refine or improve the feature with your
+feedback.
+
+Here is a quick overview of what you can do with dynamic routing:
+
+- maintain thimprovement to ex, and slower ones in a second batch if needed
```

Comment:
I think the text is cutoff here `maintain thimprovement to ex`

Reply:
Oopsie :angel: 

A last minute bad move…